assets/js/modules/analytics-4/hooks/useAudienceTilesReports.js

```diff
@@ -0,0 +1,283 @@
+/**
+ * Analytics useAudienceTilesReports custom hook.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { useInViewSelect, useSelect } from 'googlesitekit-data';
+import { CORE_USER } from '../../../googlesitekit/datastore/user/constants';
+import { DATE_RANGE_OFFSET, MODULES_ANALYTICS_4 } from '../datastore/constants';
+
+export default function useAudienceTilesReports() {
+	// An array of audience resource names.
+	const configuredAudiences = useSelect( ( select ) =>
+		select( CORE_USER ).getConfiguredAudiences()
+	);
+
+	const audiencesDimensionFilter = {
+		audienceResourceName: configuredAudiences,
+	};
+
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+			compare: true,
+		} )
+	);
+
+	const { startDate, endDate } = dates;
+
+	const reportOptions = {
+		...dates,
+		dimensions: [ { name: 'audienceResourceName' } ],
+		dimensionFilters: audiencesDimensionFilter,
+		metrics: [
+			{ name: 'totalUsers' },
+			{ name: 'sessionsPerUser' },
+			{ name: 'screenPageViewsPerSession' },
+			{ name: 'screenPageViews' },
+		],
+	};
+	const report = useInViewSelect( ( select ) => {
+		return select( MODULES_ANALYTICS_4 ).getReport( reportOptions );
+	} );
+	const reportLoaded = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).hasFinishedResolution( 'getReport', [
+			reportOptions,
+		] )
+	);
+	const reportError = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getErrorForSelector( 'getReport', [
+			reportOptions,
+		] )
+	);
+
+	const newVsReturningReportOptions = {
+		...dates,
+		dimensions: [ { name: 'newVsReturning' } ],
+		dimensionFilters: { newVsReturning: [ 'new', 'returning' ] },
+		metrics: [
+			{ name: 'totalUsers' },
+			{ name: 'sessionsPerUser' },
+			{ name: 'screenPageViewsPerSession' },
+			{ name: 'screenPageViews' },
+		],
+	};
+	const siteKitAudiencesReport = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( newVsReturningReportOptions )
+	);
+	const siteKitAudiencesReportLoaded = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).hasFinishedResolution( 'getReport', [
+			newVsReturningReportOptions,
+		] )
+	);
+	const siteKitAudiencesReportError = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getErrorForSelector( 'getReport', [
+			newVsReturningReportOptions,
+		] )
+	);
```

Comment:
As per my other comment, we need to ensure that either one or both of `report` and `siteKitAudiencesReport` are requested depending on the current audience selection and the SK partial data state where relevant.

Here's a quick suggestion for how this could look, please note this is not tested:

```suggestion
	const shouldFetchReport =
		isSiteKitAudiencePartialData === undefined
			? undefined
			: otherAudiences.length > 0 ||
			  isSiteKitAudiencePartialData === false;

	const shouldFetchSiteKitAudiencesReport =
		siteKitAudiences.length > 0 ? isSiteKitAudiencePartialData : false;

	const reportOptions = {
		...dates,
		dimensions: [ { name: 'audienceResourceName' } ],
		dimensionFilters: audiencesDimensionFilter,
		metrics: [
			{ name: 'totalUsers' },
			{ name: 'sessionsPerUser' },
			{ name: 'screenPageViewsPerSession' },
			{ name: 'screenPageViews' },
		],
	};
	const report = useInViewSelect( ( select ) => {
		if ( shouldFetchReport === undefined ) {
			return undefined;
		}

		if ( ! shouldFetchReport ) {
			return null;
		}

		return select( MODULES_ANALYTICS_4 ).getReport( reportOptions );
	} );
	const reportLoaded = useSelect( ( select ) => {
		if ( shouldFetchReport === undefined ) {
			return undefined;
		}

		if ( ! shouldFetchReport ) {
			// If we don't need to fetch the report, we can consider it loaded to simplify the loading state logic.
			return true;
		}

		return select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
			'getReport',
			[ reportOptions ]
		);
	} );
	const reportError = useSelect( ( select ) => {
		if ( shouldFetchReport === undefined ) {
			return undefined;
		}

		if ( ! shouldFetchReport ) {
			return null;
		}

		return select( MODULES_ANALYTICS_4 ).getErrorForSelector( 'getReport', [
			reportOptions,
		] );
	} );

	const newVsReturningReportOptions = {
		...dates,
		dimensions: [ { name: 'newVsReturning' } ],
		dimensionFilters: { newVsReturning: [ 'new', 'returning' ] },
		metrics: [
			{ name: 'totalUsers' },
			{ name: 'sessionsPerUser' },
			{ name: 'screenPageViewsPerSession' },
			{ name: 'screenPageViews' },
		],
	};
	const siteKitAudiencesReport = useInViewSelect( ( select ) => {
		if ( shouldFetchSiteKitAudiencesReport === undefined ) {
			return undefined;
		}

		if ( ! shouldFetchSiteKitAudiencesReport ) {
			return null;
		}

		return select( MODULES_ANALYTICS_4 ).getReport(
			newVsReturningReportOptions
		);
	} );
	const siteKitAudiencesReportLoaded = useSelect( ( select ) => {
		if ( shouldFetchSiteKitAudiencesReport === undefined ) {
			return undefined;
		}

		if ( ! shouldFetchSiteKitAudiencesReport ) {
			// If we don't need to fetch the report, we can consider it loaded to simplify the loading state logic.
			return true;
		}

		return select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
			'getReport',
			[ newVsReturningReportOptions ]
		);
	} );
	const siteKitAudiencesReportError = useSelect( ( select ) => {
		if ( shouldFetchSiteKitAudiencesReport === undefined ) {
			return undefined;
		}

		if ( ! shouldFetchSiteKitAudiencesReport ) {
			return null;
		}

		return select( MODULES_ANALYTICS_4 ).getErrorForSelector( 'getReport', [
			newVsReturningReportOptions,
		] );
	} );
```